src/gui/DetailsWidget.cpp

```diff
@@ -136,14 +139,16 @@ void DetailsWidget::getSelectedEntry(Entry* selectedEntry)
     }
 
     if (m_currentEntry->hasTotp()) {
-        m_ui->totpButton->show();
-        updateTotp();
-
         m_step = m_currentEntry->totpStep();
 
+        if (nullptr != m_timer) {
+            m_timer->stop();
```

Comment:
@adolfogc one last thing: shouldn't we also delete this timer before creating a new one?

Reply:
@louib This isn't necessary because its parent will delete it.

Reference:
>QObjects organize themselves in object trees. When you create a QObject with another object as parent, the object will automatically add itself to the parent's children() list. The parent takes ownership of the object; i.e., it will automatically delete its children in its destructor. You can look for an object by name and optionally type using findChild() or findChildren().

Source: http://doc.qt.io/qt-5/qobject.html

Reply:
@louib We can let its parent delete it or we could also delete it ourselves at that time to save some memory (given the possible lifetime of its parent). Is that what you meant?

Reply:
@adolfogc didn't know about the destruction behaviour of `QObject`! It's your call then.

Reply:
@louib I prefer to let QObject take care of it.

Reply:
Actually the m_timer has no parent! When you create it in the constructor you pass nullptr, you should pass "this" to establish the DetailsWidget as its parent.